meson.build

```diff
@@ -0,0 +1,1823 @@
+project('mpv',
+        'c',
+        license: ['GPL2+', 'LGPL2.1+'],
+        version: files('./VERSION'),
+        meson_version: '>=0.60.0',
+        default_options: [
+            'buildtype=debugoptimized',
+            'b_lundef=false',
+            'c_std=c11',
+            'warning_level=1',
+        ]
+)
+
+build_root = meson.project_build_root()
+source_root = meson.project_source_root()
+
+pymod = import('python')
+python = pymod.find_installation('python3')
```

Comment:
You don't need this, it is effectively identical to `find_program('python3')`.

The purpose of the python module's find_installation() method is to also add additional contexts to this found program, which are used when compiling python C extensions or installing python  modules to the system site-packages/. You're not using any of that context, just the found program itself as a command.

Reply:
Thanks for the comment, I just noticed the other night with libplacebo that `import('python').find_installation()` actually ended up finding the binary distribution of meson on macOS as the "python" if installed, so I will link this that way as well, as it's being utilized to run a single binary with the correct interpreter as well.

(Xcode does come with python3(.8) in PATH so you can guess my late-night face when I noticed meson was finding `meson` as the python interpreter :) )

Reply:
`meson runpython` can be used to run python scripts in an emergency, which is handy if you have meson installed on an OS that doesn't have a python installed -- but does have python bundled in meson itself, as is the case for at least the Windows .msi installer.

If `find_program('python3')` doesn't find a python3 command anywhere it knows to look (machine file overrides, PATH lookup) it will special case that one program lookup and return either `[sys.executable]` (the python that meson itself is running on) or, if that would be meson.exe, it returns `[sys.executable, 'runpython']`.

find_installation will do the same thing if you don't specify either python2 or python3, on the assumption that people who care about building extension modules and installing python sources to site-packages will surely specify whether they want to do that for python2 or python3... it's oddly common for people to use find_installation just so they can use it like find_program and run scripts.